src/date-picker/date-picker.scss

```diff
@@ -14,117 +14,104 @@
+    align-items: center;
+    justify-content: center;
+    line-height: $iui-baseline * 4;
+  }
 
-        .iui-month {
-          overflow: hidden;
-          max-width: 15ch;
-          text-overflow: ellipsis;
-          white-space: nowrap;
-          flex-shrink: 0;
-        }
-      }
+  .iui-month {
+    overflow: hidden;
+    max-width: 15ch;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+    flex-shrink: 0;
+  }
+}
 
-      > .iui-weekdays {
-        line-height: $iui-baseline * 3;
-        display: flex;
-        justify-content: space-evenly;
-        align-items: center;
-        padding: 0 $iui-s;
-        @include themed {
-          background-color: t(iui-color-background-2);
-        }
+@mixin iui-calendar-weekdays {
+  line-height: $iui-baseline * 3;
+  display: flex;
+  justify-content: space-evenly;
+  align-items: center;
+  padding: 0 $iui-s;
+  font-weight: $iui-font-weight-bold;
+  @include themed {
+    background-color: t(iui-color-background-2);
+  }
 
-        > div {
-          white-space: nowrap;
-          overflow: hidden;
-          width: $iui-xl;
-          text-align: center;
-        }
-      }
-    }
+  > div {
+    white-space: nowrap;
+    overflow: hidden;
+    width: $iui-xl;
+    text-align: center;
+  }
+}
 
-    > .iui-dates {
-      > .iui-week {
-        white-space: nowrap;
-        height: $iui-baseline * 3;
-        display: flex;
-        align-items: center;
-        justify-content: space-evenly;
-        padding: 0 $iui-s;
+@mixin iui-calendar-week {
+  white-space: nowrap;
+  height: $iui-baseline * 3;
+  display: flex;
+  align-items: center;
+  justify-content: space-evenly;
+  padding: 0 $iui-s;
 
-        > .iui-date {
-          @include iui-focus;
-          cursor: pointer;
-          display: inline-block;
-          width: $iui-baseline * 3;
-          line-height: $iui-baseline * 3;
+  > .iui-date {
+    @include iui-focus;
+    cursor: pointer;
+    display: inline-block;
+    width: $iui-baseline * 3;
+    line-height: $iui-baseline * 3;
 
-          &:hover:not(.iui-selected) {
-            border-radius: $iui-border-radius;
-            font-weight: $iui-font-weight-semibold;
-            @include themed {
-              color: t(iui-color-foreground-primary);
-              background-color: rgba(t(iui-color-foreground-primary-rgb), t(iui-opacity-6));
-            }
-          }
+    &.iui-today {
+      font-weight: $iui-font-weight-semibold;
+      border-radius: 50%;
+      @include themed {
+        background-color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-6));
+      }
+    }
 
-          &.iui-today {
-            font-weight: $iui-font-weight-semibold;
-            border-radius: 50%;
-            @include themed {
-              background-color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-6));
-            }
-          }
+    &:hover {
+      border-radius: $iui-border-radius;
+      font-weight: $iui-font-weight-semibold;
```

Comment:
@FlyersPh9 We could take this opportunity to rethink/improve hover styles.

The current styling is weird for two reasons:
- current date changes shape for some reason
- other dates have content shift (because of font-weight change)

Reply:
Would you mind doing this with separate PR? :D